src/main/java/lotto/util/LottoManager.java

```diff
@@ -0,0 +1,143 @@
+package lotto.util;
+
+import static lotto.numbers.Lotto.MAX_LOTTO_NUMBER;
+import static lotto.numbers.Lotto.MIN_LOTTO_NUMBER;
+import static lotto.numbers.Lotto.SIZE_OF_LOTTO;
+import static lotto.util.Money.LOTTO_PRICE;
+
+import camp.nextstep.edu.missionutils.Console;
+import camp.nextstep.edu.missionutils.Randoms;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lotto.numbers.Lotto;
+import lotto.numbers.UserInputNumbers;
+
+public class LottoManager {
+
+  public static final int PERCENTAGE_FACTOR = 100;
+
+  public void runLotto() {
+    Money money = receivePurchaseAmount();
+    List<Lotto> lottos = generateLottos(money.getAmount());
+    printGeneratedLottoNumbers(lottos);
+    UserInputNumbers receivedLotto = receiveLottoNumber();
+    Map<WinningCheck, Integer> result = calculateWinningResult(lottos, receivedLotto);
+    printWinningResult(result);
+    printProfitRate(money.getAmount(), result);
+  }
+
+  private Money receivePurchaseAmount() {
+    Money money;
+    while (true) {
+      try {
+        System.out.println("구입금액을 입력해 주세요.");
+        String input = Console.readLine();
+        money = new Money(input);
+        break;
+      } catch (IllegalArgumentException e) {
+        System.out.println(e.getMessage());
+      }
+    }
+    return money;
+  }
+
+  private List<Lotto> generateLottos(int money) {
+    int numberOfLotto = money / LOTTO_PRICE;
+    System.out.println("\n" + numberOfLotto + "개를 구매했습니다.");
+
+    List<Lotto> lottos = new ArrayList<>();
+
+    for (int i = 0; i < numberOfLotto; i++) {
+      List<Integer> lottoNumbers =
+          Randoms.pickUniqueNumbersInRange(MIN_LOTTO_NUMBER, MAX_LOTTO_NUMBER, SIZE_OF_LOTTO);
+
+      try {
+        Lotto lotto = new Lotto(lottoNumbers);
+        lottos.add(lotto);
+      } catch (IllegalArgumentException e) {
+        System.out.println(e.getMessage());
+        i--;
+      }
+    }
+
+    return lottos;
+  }
+
+  private void printGeneratedLottoNumbers(List<Lotto> lottos) {
+    for (Lotto lotto : lottos) {
+      List<Integer> lottoNumbers = new ArrayList<>(lotto.getLottoNumbers());
+      Collections.sort(lottoNumbers);
+      System.out.println(lottoNumbers);
+    }
+  }
+
+  private UserInputNumbers receiveLottoNumber() {
+    while (true) {
+      System.out.println("\n당첨 번호를 입력해 주세요.");
+      String inputNumbers = Console.readLine();
+      try {
+        String[] numbersArray = inputNumbers.split(",");
+        List<Integer> receivedLottoNumbers = new ArrayList<>();
+        Arrays.stream(numbersArray)
+            .forEach(number -> receivedLottoNumbers.add(Integer.parseInt(number.trim()))); // 공백 제거
```

Comment:
```suggestion
        List<Integer> receivedLottoNumbers = Arrays.stream(numbersArray)
        .map(String::trim)
        .map(Integer::parseInt)
        .toList();
```
처럼 스트림으로 한 번에 collect 해줄수도 있습니다

Reply:
Stream을 생성한 후 map()을 두 번 사용해서 공백 제거와 정수 변환을 순차적으로 한 뒤에 다시 리스트로 변환시켰네요.
이 방법으로 바꿔보겠습니다. 감사합니다.

추가로 Java 16부터 .toList() 만으로 리스트 변환이 가능하다고 합니다.
이전 버전에서는 .collect(Collectors.toList()) 이렇게 변환해야합니다.